_posts/2023-02-01-user-defined-operators.md

```diff
@@ -0,0 +1,237 @@
+
+### Introducing User-Defined Operators
+The new feature adds another form of the `using for` syntax, where instead of attaching a function,
+it is assigned an operator symbol.
+The function is then invoked whenever the operator is used on the values of the type:
+
+```solidity
+type Int is int;
+using {add as +} for Int;
+
+function add(Int a, Int b) pure returns (Int) {
+    return Int.wrap(Int.unwrap(a) + Int.unwrap(b));
+}
+
+function test(Int a, Int b) pure returns (Int) {
+    return a + b;
+}
+```
+
+Note that defining an operator for a type does not automatically make it available as an attached function.
+E.g. `using {add as +} for Int` does not imply `using {add} for Int`.
+
+#### But Why?
+The primary use case we had in mind when extending UDVTs was support for fixed-point numbers.
+
+Fixed-point types were originally going to be an integral part of the language, and the
+partially implemented [`ufixed` and `fixed`](https://docs.soliditylang.org/en/latest/types.html#fixed-point-numbers)
+types are the vestiges of that old design.
+While trying to agree the final design for the feature, however, we realized that no implementation
+is going to be a trade-off good enough for most situations and cannot become a suitable default choice.
+We decided to leave the choice up to library authors and instead focus on providing primitives
+necessary to build their own fixed-point types that feel like a native type.
+User-defined operators are a step towards that goal.
+
+#### Which Operators Can Be User-Defined?
+| Category   | Operators                        |
+|------------|----------------------------------|
+| Bitwise    | `&`, `\|`, `^`,  `~`             |
+| Arithmetic | `+`, `-`, `*`, `/`, `%`          |
+| Comparison | `==`, `!=`, `<`, `<=`, `>`, `>=` |
+
+Of the operators listed above, `~` is unary, `-` can be both unary and binary and the rest is always binary.
+Note that Solidity does not have binary `+`.
+
+#### Which Types Can Have User-Defined Operators?
+Currently, operators can only be defined for user-defined value types.
+We are also considering enabling them for structs and enums in future releases.
+There are no plans, for now, to allow defining operators for built-in value types, especially in cases where
+an operator is already provided by the compiler.
+
+Note that the type wildcard (`*`) is not allowed in `using for` when defining operators, which
+is in line with how it already worked when used to attach individual functions.
+
+Since currently there are no implicit conversions allowed to and from UDVTs, user-defined operators
+must always be invoked with exact types.
+
+#### What Are the Restrictions on User-Defined Operator Definitions?
+The operator definition must be a pure function, defined at the file level or a library.
+All parameters must be of the same type, which must be the type specified in `using for`.
```

Comment:
```suggestion
All parameters must be of the same type specified in `using for`.
```

Reply:
I'll try to reword it, but the proposed version sounds a bit ambiguous to me. As if the parameters were defined in `using for` :)